remotespark/kernels/kernelmagics.py

```diff
@@ -206,23 +211,38 @@ def delete(self, line, cell="", local_ns=None):
                 self.ipython_display.send_error("Cannot delete this kernel's session ({}). Specify a different session,"
                                                 " shutdown the kernel to delete this session, or run %cleanup to "
                                                 "delete all sessions for this endpoint.".format(id))
-                return
+                return None
 
             self.spark_controller.delete_session_by_id(self.connection_string, session)
         else:
             self.ipython_display.send_error("Include the -f parameter if you understand that all statements executed"
                                             "in this session will be lost.")
-            return
+            return None
 
     @cell_magic
     def _do_not_call_start_session(self, line, cell="", local_ns=None):
-        if not self.session_started:
-            self.session_started = True
+        # Starts a session unless session is already created or a fatal error occurred. Returns True when session is
+        # created successfully.
 
+        if self.fatal_error:
+            self.ipython_display.send_error(self.fatal_error_message)
+            return False
+
+        if not self.session_started:
             skip = False
             properties = conf.get_session_properties(self.language)
 
-            self.spark_controller.add_session(self.session_name, self.connection_string, skip, properties)
+            try:
+                self.session_started = True
+                self.spark_controller.add_session(self.session_name, self.connection_string, skip, properties)
+            except Exception as e:
```

Comment:
What's "PEP"? Generally, this will help the code reader to identify whether this is a different statement separated from the try or not.